core-web/libs/sdk/react/src/lib/components/DotEditableText/DotEditableText.tsx

```diff
@@ -0,0 +1,135 @@
+import { Editor } from '@tinymce/tinymce-react';
+import { useEffect, useRef, useState } from 'react';
+
+import {
+    isInsideEditor as isInsideEditorFn,
+    postMessageToEditor,
+    CUSTOMER_ACTIONS,
+    DotCmsClient
+} from '@dotcms/client';
+
+import { DotEditableTextProps, TINYMCE_CONFIG } from './utils';
+
+export function DotEditableText({
+    mode = 'plain',
+    format = 'text',
+    contentlet,
+    fieldName = ''
+}: Readonly<DotEditableTextProps>) {
+    const editorRef = useRef<Editor['editor'] | null>(null);
+    const [isInsideEditor, setIsInsideEditor] = useState(false);
+    const [content, setContent] = useState(contentlet[fieldName] || '');
+
+    useEffect(() => {
+        const isInsideEditor = isInsideEditorFn();
+        setIsInsideEditor(isInsideEditor);
+
+        if (!isInsideEditor) {
+            return;
+        }
+
+        const content = contentlet[fieldName] || '';
+        editorRef.current?.setContent(content, { format });
+        setContent(content);
+    }, [contentlet, fieldName, format]);
```

Comment:
Because the `contentlet` is an object that change refs here, are we not getting extra re-runs we might not want?

Reply:
Ok